ngraph/test/type_prop/einsum.cpp

```diff
@@ -0,0 +1,291 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "gtest/gtest.h"
+#include "ngraph/ngraph.hpp"
+#include "util/type_prop.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+TEST(type_prop, einsum_staticshape_dotproduct)
+{
+    std::string equation = "i,i->";
+    Shape input1_shape{3};
+    Shape input2_shape{3};
+    Shape out_shape{};
+    auto I1 = make_shared<op::Parameter>(element::f32, input1_shape);
+    auto I2 = make_shared<op::Parameter>(element::f32, input2_shape);
+    auto O = make_shared<op::v7::Einsum>(OutputVector{I1, I2}, equation);
+    ASSERT_EQ(O->get_element_type(), element::f32);
+    ASSERT_EQ(O->get_shape(), out_shape);
+}
+
+TEST(type_prop, einsum_staticshape_matmul)
+{
+    std::string equation = "ab,bc->ac";
+    Shape input1_shape{2, 3};
+    Shape input2_shape{3, 4};
+    Shape out_shape{2, 4};
+    auto I1 = make_shared<op::Parameter>(element::f32, input1_shape);
+    auto I2 = make_shared<op::Parameter>(element::f32, input2_shape);
+    auto O = make_shared<op::v7::Einsum>(OutputVector{I1, I2}, equation);
+    ASSERT_EQ(O->get_element_type(), element::f32);
+    ASSERT_EQ(O->get_shape(), out_shape);
+}
+
+TEST(type_prop, einsum_staticshape_trace)
+{
+    std::string equation = "kii->k";
+    Shape input1_shape{2, 3, 3};
+    Shape out_shape{2};
+    auto I1 = make_shared<op::Parameter>(element::f32, input1_shape);
+    auto O = make_shared<op::v7::Einsum>(OutputVector{I1}, equation);
+    ASSERT_EQ(O->get_element_type(), element::f32);
+    ASSERT_EQ(O->get_shape(), out_shape);
+}
+
+TEST(type_prop, einsum_staticshape_diagextraction)
+{
+    std::string equation = "kii->ki";
+    Shape input1_shape{2, 3, 3};
+    Shape out_shape{2, 3};
+    auto I1 = make_shared<op::Parameter>(element::f32, input1_shape);
+    auto O = make_shared<op::v7::Einsum>(OutputVector{I1}, equation);
+    ASSERT_EQ(O->get_element_type(), element::f32);
+    ASSERT_EQ(O->get_shape(), out_shape);
+}
+
+TEST(type_prop, einsum_staticshape_transpose)
+{
+    std::string equation = "ijk->kij";
+    Shape input1_shape{1, 3, 3};
```

Comment:
let's change the input shape to be all distinct non-1 elements. For example, `{2, 3, 5}`

Reply:
Fixed.